### DISCUSSION TITLE
Function Redis::setTimeout() is deprecated in Unknown on line 0

### DISCUSSION THREAD

**Discussion by chengpeiyang** [phalcon/cphalcon#15312] [General]
Function Redis::setTimeout() is deprecated in Unknown on line 0
php:7.2.34
phpredis: 5.3.2
phalcon:3.4.3 or 3.4.4

How can I fix this deprecated warning?  I don't want to downgrade phpredis to 4.x. thanks!

**Comment by pgrimaud** [comment]
This warning has been fixed on https://github.com/phalcon/cphalcon/tree/3.4.x but not on phalcon 3.4.5

Refs to https://github.com/phalcon/cphalcon/issues/14281#issuecomment-746178895
> you need to make own build from 3.4.x branch - https://github.com/phalcon/cphalcon/tree/3.4.x
> See, for more details - https://docs.phalcon.io/3.4/en/installation#compile-from-sources

**Comment by Jeckerson** [comment]
Hello, version `3.4` is deprecated and not supported from our side. As mentioned @pgrimaud you can pick branch `3.4.x` and compile by yourself the extension with fix.

**Comment by noerarief23** [comment]
You can downgrade php-redis https://pecl.php.net/package/redis/4.2.0/windows
